lua/entities/kleiner/init.lua

```diff
@@ -0,0 +1,916 @@
+            path:Update(self)
+        end
+
+        if (options.draw) then
+            path:Draw()
+        end
+
+        if (self.loco:IsStuck()) then
+            self:HandleStuck()
+
+            return "stuck"
+        end
+
+        coroutine.yield()
+    end
+
+    return "ok"
+end
+
+--this function runs during the path movement. returning false will interrupt path movement, in case other actions are needed.
+function ENT:WhilePathing(path)
+    if (path == nil or not path:IsValid()) then return true end
+    if (self.loco == nil) then return true end
+    if (not self:OnGround() and self:GetVelocity().z < -50) then return false end --attempting to move while falling seems to pause falling
+    local seg1, index = self:GetCurrentPathPoint() --this returns the first path segment we're closest to.
+    local seg2 = path:GetAllSegments()[index + 1]
+    local seg3 = path:GetAllSegments()[index + 2]
+    local dir = self:GetForward()
+
+    if (seg1 and seg2) then
+        dir = (seg2.pos - seg1.pos):GetNormalized()
+    end
+
+    -- a quick fix to the broken ladder handling. If they are about to climb a ladder, teleport them instead
+    --everything goes wrong when we use ladders so let's try to teleport over them.
+    if (seg2 and seg2.ladder:IsValid()) then
+        self:Teleport((seg3 and seg3.pos) or (seg2 and seg2.pos))
+        self.loco:ClearStuck()
+
+        return
+    end
+
+    --If any nav areas are marked as STOP and AVOID, we automatically teleport over them.
+    if (seg2 and seg2.area:HasAttributes(NAV_MESH_JUMP) and seg2.area:HasAttributes(NAV_MESH_AVOID)) then
+        --self:Teleport((seg3 and seg3.pos) or (seg2 and seg2.pos))
+        if (seg3 and seg3.area ~= seg1.area) then
+            self.loco:JumpAcrossGap((seg3 and seg3.pos) or (seg2 and seg2.pos), dir)
+            self.loco:ClearStuck()
+
+            return
+        end
+    end
+
+    -- if they're on a ladder, their physics have been permanently broken so we'll just delete him.
+    if (self.loco:IsUsingLadder()) then
+        self:Remove()
+    end
+
+    -- Jumping handling
+    local ofs = (seg2.pos - self:GetPos())
+    local heightdist = ofs.z
+    local lendist = (ofs * Vector(1, 1, 0)):Length()
+    local jumpdir = (ofs * Vector(1, 1, 0)):GetNormalized()
+    local inrange = lendist < 64
+    local shouldjump = inrange and (seg3 and (seg3.area:HasAttributes(NAV_MESH_JUMP) or seg3.type == 2 or seg3.type == 3))
+    local across = (math.abs(heightdist) < 64)
+
+    if (self:IsOnGround() and shouldjump) then
+        if (across) then
+            self.loco:JumpAcrossGap((seg3 and seg3.pos) or (seg2 and seg2.pos), dir)
+        else
+            self.loco:Jump()
+        end
+    end
+
+    return true
+end
+
+ENT.PathGen = function(area, fromArea, ladder, elevator, length)
+    KLPATHGEN_ITERS = (KLPATHGEN_ITERS or 0) + 1
+    local self = RELEVANT_KLEINER --this is bullshit, i guess this callback doesn't include the entity pathing. 
```

Comment:
You could also put this function inside ENT:ChaseTarget to capture the local `self` variable